task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -547,7 +547,7 @@ class CommsDecoder(Generic[ReceiveMsgType, SendMsgType]):
     # "sort of wrong default"
     decoder: TypeAdapter[ReceiveMsgType] = attrs.field(factory=lambda: TypeAdapter(ToTask), repr=False)
 
-    lock: threading.Lock = attrs.field(factory=threading.Lock, repr=False)
+    lock: aiologic.Lock = attrs.field(factory=aiologic.Lock, repr=False)
```

Comment:
We should probably add a comment here that this is just meant to be used with async

Reply:
TIL!